apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java

```diff
@@ -338,6 +344,29 @@ static AppConfiguration readConfigFromEnvironmentVariables() {
               "Missing value for configuration environment variable '%s'.",
               ENV_VAR_KEY_DATABASE_PASSWORD));
 
+    String databaseMaxPoolSizeText = System.getenv(ENV_VAR_KEY_DATABASE_MAX_POOL_SIZE);
+    Optional<Integer> databaseMaxPoolSizeOption;
+    if (databaseMaxPoolSizeText == null || databaseMaxPoolSizeText.isEmpty()) {
+      databaseMaxPoolSizeOption = Optional.empty();
+    } else {
+      try {
+        databaseMaxPoolSizeOption = Optional.of(Integer.parseInt(databaseMaxPoolSizeText));
+      } catch (NumberFormatException e) {
+        databaseMaxPoolSizeOption = Optional.of(-1);
+      }
+      if (databaseMaxPoolSizeOption.get() < 1)
+        throw new AppConfigurationException(
+            String.format(
+                "Invalid value for configuration environment variable '%s': '%s'",
+                ENV_VAR_KEY_DATABASE_MAX_POOL_SIZE, databaseMaxPoolSizeText));
+    }
+
+    /*
+     * Note: For CcwRifLoadJob, databaseMaxPoolSize needs to be double the number of loader threads
+     * when idempotent loads are being used. Apparently, the queries need a separate Connection?
+     */
+    Integer maxPoolSize = databaseMaxPoolSizeOption.orElse(loaderThreads * 2);
+
```

Comment:
Moving the rdaLoadOptions initialization above this would allow the default to be increased by 2 if the RDA pipeline jobs have been enabled (rdaLoadOptions != null).

Reply:
I'd suggest making that change in a separate DC-Geo specific PR, just for tracking purposes.